sound/soc/sof/sof-audio.c

```diff
@@ -58,8 +57,6 @@ static int sof_widget_free_unlocked(struct snd_sof_dev *sdev,
 	if (--swidget->use_count)
 		return 0;
 
-	pipe_widget = swidget->spipe->pipe_widget;
```

Comment:
it is not just the oopses, but what about the core_put and the other things in this path?
Originally this code was 'detecting' the DSPless mode with the `swidget->private == NULL` check since we only create swidget for DAIs and we did not store anything in private.

In any case, this patch should be moved earlier, before saving sdai to swidget->private.

Reply:
I am not able to follow any of the comments for this patch @ujfalusi 

for the DSPless mode, the core_put() is irrelevant. There is no DSP used so no cores.

In addition, the only problem that I tried to fix is the oopses. Nothing more.

I did move the patch earlier but not sure what else you are suggesting.

Reply:
this patch is useless anyways, I am not sure why it was added. Maybe an earlier version. Removed now